docs/specs/om/open_metrics_spec_2_0.md

```diff
@@ -216,42 +226,103 @@ MetricFamilies of type Info MUST have an empty Unit string.
+
+If the zero bucket is present, any measured value that falls into the zero bucket MUST be counted towards the zero bucket and MUST NOT be counted in any other exponential bucket. The Zero threshold SHOULD be equal to a lower limit of an arbitraty exponential bucket.
+
+If the NaN value is not allowed, then the Count value MUST be equal to the sum of the negative, positive and zero buckets.
+
+If the NaN value is allowed, it SHOULD NOT be counted in any bucket. The value NaN MAY be counted in the +Inf bucket, and MUST not be counted in any other bucket. In case the NaN is counted in the +Inf bucket, then the Count value MUST be equal to the sum of the negative, positive and zero buckets, otherwise the Count MUST be greater. The ratonale is that NaN does not belong to any bucket mathematically, and instrumentation libraries traditionally don't put it into any bucket, which is why the wording "SHOULD NOT" is used.
 
-Each bucket covers the values less and or equal to it, and the value of the exemplar MUST be within this range. Exemplars SHOULD be put into the bucket with the highest value. A bucket MUST NOT have more than one exemplar.
+A Histogram MetricPoint with exponential buckets MAY contain exemplars.
+
+Exemplars associated with a Histogram MetricPoint with exponential buckets MUST have a timestamp.
+
+The values of exemplars in a Histogram MetricPoint with exponential buckets MUST fall into one of the exponential buckets.
+
+The values of exemplars in a Histogram MetricPoint with exponential buckets SHOULD be evenly distributed to avoid only representing the bucket with the highest value and therefore most common case.
 
 #### GaugeHistogram
 
 GaugeHistograms measure current distributions. Common examples are how long items have been waiting in a queue, or size of the requests in a queue.
 
-A GaugeHistogram MetricPoint MUST have one bucket with an +Inf threshold, and SHOULD contain a Gsum value. Every bucket MUST have a threshold and a value.
+A GaugeHistogram MetricPoint MUST contain either classic buckets or exponential buckets or both.
 
-The buckets for a GaugeHistogram follow all the same rules as for a Histogram.
+A GaugeHistogram MetricPoint SHOULD contain Gcount, Gsum. Every bucket MUST have well defined boundaries and a value. Boundaries of a bucket MUST NOT be NaN. Gcount and bucket values MUST be integers.
 
-The bucket and Gsum of a GaugeHistogram are conceptually gauges, however bucket values MUST NOT be negative or NaN. If negative threshold buckets are present, then sum MAY be negative. Gsum MUST NOT be NaN. Bucket values MUST be integers.
+The bucket and Gsum of a GaugeHistogram are conceptually gauges, however bucket values MUST NOT be negative or NaN. If negative threshold buckets are present, then Gsum MAY be negative. Gsum MUST NOT be NaN. Bucket values MUST be integers.
```

Comment:
I think we should drop "If negative threshold buckets are present, then Gsum MAY be negative." This makes classic histograms without negative bucket threshold but negative observations unrepresentable for no reason.

"Gsum MUST NOT be NaN." is also problematic because this is the case if NaN observations are in the histogram, and therefore those histograms are unrepresantable.